infra/version_check.py

```diff
@@ -5,6 +5,13 @@
 
 import subprocess
 
+
+class NotFoundError(Exception):
+    def __init__(self, notfound):
+        message = f"Unexpected {notfound} notfound"
+        super().__init__(message)
+
+
```

Comment:
I do not think the name `NotFoundError` is a good fit. Also, I do not see why having `__init__()` here.

What will the exception message look like here?

Reply:
class UnexpectedValueError(ValueError):
    def __init__(self, value):
        message = f"Unexpected value: {value}"
        super().__init__(message)

raise UnexpectedValueError("Failure while getting latest tag")

Exception message: "Unexpected value: Failure while getting latest tag".

Is this okay??
@suhaibmujahid

Reply:
I guess we could use `RuntimeError` here instead of creating a custom exception